bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/handler/ForecastSolarBridgeHandler.java

```diff
@@ -0,0 +1,226 @@
+        PointType locationConfigured;
+
+        // handle location error cases
+        if (config.location.isBlank()) {
+            if (homeLocation.isEmpty()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/solarforecast.site.status.location-missing");
+                return;
+            } else {
+                locationConfigured = homeLocation.get();
+                // continue with openHAB location
+            }
+        } else {
+            try {
+                locationConfigured = new PointType(config.location);
+                // continue with location from configuration
+            } catch (Exception e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                return;
+            }
+        }
+        Configuration editConfig = editConfiguration();
+        editConfig.put("location", locationConfigured.toString());
+        updateConfiguration(editConfig);
+        config = getConfigAs(ForecastSolarBridgeConfiguration.class);
+        configuration = Optional.of(config);
+        updateStatus(ThingStatus.UNKNOWN);
+        refreshJob = Optional
+                .of(scheduler.scheduleWithFixedDelay(this::getData, 0, REFRESH_ACTUAL_INTERVAL, TimeUnit.MINUTES));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            String channel = channelUID.getIdWithoutGroup();
+            switch (channel) {
+                case CHANNEL_ENERGY_ACTUAL:
+                case CHANNEL_ENERGY_REMAIN:
+                case CHANNEL_ENERGY_TODAY:
+                case CHANNEL_POWER_ACTUAL:
+                    getData();
+                    break;
+                case CHANNEL_POWER_ESTIMATE:
+                case CHANNEL_ENERGY_ESTIMATE:
+                    forecastUpdate();
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Get data for all planes. Synchronized to protect plane list from being modified during update
+     */
+    private synchronized void getData() {
+        if (planes.isEmpty()) {
+            return;
+        }
+        boolean update = true;
+        double energySum = 0;
+        double powerSum = 0;
+        double daySum = 0;
+        for (Iterator<ForecastSolarPlaneHandler> iterator = planes.iterator(); iterator.hasNext();) {
+            try {
+                ForecastSolarPlaneHandler sfph = iterator.next();
+                ForecastSolarObject fo = sfph.fetchData();
+                ZonedDateTime now = ZonedDateTime.now(fo.getZone());
+                energySum += fo.getActualEnergyValue(now);
+                powerSum += fo.getActualPowerValue(now);
+                daySum += fo.getDayTotal(now.toLocalDate());
+            } catch (SolarForecastException sfe) {
+                logger.info("{}", sfe.getMessage());
```

Comment:
Please see comment for the other bridge handler, perhaps Thing status should be set to **OFFLINE** here.

Reply:
Exception message shown as parameter in ThingStatus - same pattern as in `SolcastBridgeHandler`